fulfillments.go

```diff
@@ -208,3 +206,56 @@ func BuildFulfillmentDTOs(fulfillments []shopify.Fulfillment) FulfillmentDTOs {
 
 	return dtos
 }
+
+// TrackingInfoDTO represents Shopify fulfillment tracking info in HTTP requests and responses
+type TrackingInfoDTO struct {
+	Company string `json:"company,omitempty"`
+	Number  string `json:"number,omitempty"`
+	URL     string `json:"url,omitempty"`
+}
+
+// BuildTrackingInfoDTO converts fulfillment tracking info into a DTO equivalent
+func BuildTrackingInfoDTO(fulfillment shopify.Fulfillment) *TrackingInfoDTO {
+	trackingNumber := ""
+	if len(fulfillment.TrackingNumbers) > 0 {
```

Comment:
Should we be checking that 0 exists in case of offset errors?

Reply:
As in replacing this with a for loop that returns the first element where `ele != ""`? The length check is good for telling us that 0 exists, just not that it's a non-empty string.

Reply:
I mean it implies there are multiple tracking numbers so we always get the first. So what happens if there is many of them?

Reply:
So it's a bit of weird one having just compared between old and new APIs. It looks like from `2022-07` onwards it's not possible to create/update a fulfillment with multiple tracking numbers/URLs at all! Despite the fact that both plural and singular fields still appear on responses from the newer version:
```json
"tracking_number": "GB123234234TX",
"tracking_numbers": [
    "GB123234234TX",
    "HB123234234TX"
],
"tracking_url": null,
"tracking_urls": []
```

Shopify always takes the first element of the plural field when populating its singular field in GET responses. I guess we could implement these singular fields `TrackingNumber`/`TrackingURL` into `shopify` and use them here instead?

Reply:
To clarify, for some odd reason it's only possible to do [multiple tracking numbers with the GraphQL API](https://shopify.dev/docs/api/admin-graphql/2022-07/mutations/fulfillmentTrackingInfoUpdateV2#returns) :roll_eyes:. I tried adding them anyway to the JSON payload in the REST API and it just ignores them :cry:.

Reply:
Yeh, that's what we should do. Shopify responses and requests, although mostly similar, do have subtle differences. If they use the tracking_number property when creating then I think we should use that. Yeh same for URL also.